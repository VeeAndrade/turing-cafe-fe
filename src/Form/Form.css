.reservation-form {
  display: flex;
  margin-bottom: 40px;
  width: 100%;
  align-self: center;
  justify-content: space-evenly;
}

.name-input,
.date-input,
.time-input,
.numOfGuest-input {
  width: 15%;
  height: 30px;
  padding-left: 10px;
}

.make-reserveration-btn {
  background-color: teal;
  color: #fff;
  width: 15%;
}