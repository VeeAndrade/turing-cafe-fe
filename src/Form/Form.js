import React, { Component } from 'react';
import './Form.css'

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      numOfGuests: ''
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  makeReservation = (e) => {
    e.preventDefault()
    const reservation = {...this.state, id: Date.now()}
    const { postReservation } = this.props
    postReservation(reservation);
    this.resetInputs()
  }

  resetInputs = () => {
    this.setState({
      name: '',
      date: '',
      time: '',
      numOfGuests: ''
    })
  }

  render() {
    return (
      <form className='reservation-form'>
        <input className='name-input'
        type='text'
        name='name'
        placeholder='Name'
        value={this.state.name}
        onChange={this.handleChange}
        />
        <input className='date-input'
        type='text'
        name='date'
        placeholder='Date (mm/dd)'
        value={this.state.date}
        onChange={this.handleChange}
        />
        <input className='time-input'
        type='text'
        name='time'
        placeholder='Time'
        value={this.state.time}
        onChange={this.handleChange}
        />
        <input className='numOfGuest-input'
        type='number'
        name='numOfGuests'
        placeholder='Number of guests'
        value={this.state.numOfGuests}
        onChange={this.handleChange}
        />
        <button onClick={this.makeReservation}className='make-reserveration-btn'>Make Reservation</button>
      </form>
    )
  }
}

export default Form;