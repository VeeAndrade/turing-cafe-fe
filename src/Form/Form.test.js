import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper
  const mockPostReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow( <Form postReservation={mockPostReservation} /> )

  })
  it('should match the snapshot with all the correct args', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = {
      target: {
        name: 'name',
        value: 'Vee'
      }
    }
    const mockEvent2 = {
      target: {
        name: 'date',
        value: '06/21'
      }
    }  

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('name')).toEqual('Vee'); 

    wrapper.instance().handleChange(mockEvent2)
    expect(wrapper.state('date')).toEqual('06/21');
  });

  it('should call make reservation and reset inputs', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().resetInputs = jest.fn;

    wrapper.instance().makeReservation(mockEvent)

    expect(mockPostReservation).toHaveBeenCalled();
  })
});