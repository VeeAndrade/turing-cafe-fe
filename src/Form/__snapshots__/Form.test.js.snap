// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all the correct args 1`] = `
<form
  className="reservation-form"
>
  <input
    className="name-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="date-input"
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    className="time-input"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    className="numOfGuest-input"
    name="numOfGuests"
    onChange={[Function]}
    placeholder="Number of guests"
    type="number"
    value=""
  />
  <button
    className="make-reserveration-btn"
    onClick={[Function]}
  >
    Make Reservation
  </button>
</form>
`;
